src/wolfsftp.c

```diff
@@ -1577,6 +1553,31 @@ int wolfSSH_SFTP_CreateStatus(WOLFSSH* ssh, word32 status, word32 reqId,
 }
 
 
+/*
+ * This is a wrapper around the function wolfSSH_RealPath. Since it modifies
+ * the source path value, copy the path from the data stream into a local
+ * array and use that as the source.
+ *
+ * @param defaultPath pointer to the defaultPath
+ * @param data        input data stream at the location of the path name
+ * @param sz          size of the path name in bytes
+ * @param s           destination buffer for the Real Path
+ * @param sSz         size of s in bytes
+ * @return            0 for success or negative error code
+ */
+static int GetAndCleanPath(const char* defaultPath,
+        const byte* data, word32 sz, char* s, word32 sSz)
+{
+    char r[WOLFSSH_MAX_FILENAME];
+
+    if (sz > sizeof r) sz = sizeof r;
```

Comment:
Shouldn't this be an error? Why truncate?

Reply:
This should be an error.